stock_available_immediately/tests/test_stock_available_immediately.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2014 Camptocamp, Akretion, Numérigraphe
+# Copyright 2016 Sodexis
+# Copyright 2019 Sergio Díaz <sergiodm.1989@gmail.com>
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestStockLogisticsWarehouse(TransactionCase):
+    def test01_stock_levels(self):
+        """
+        Checking that immediately_usable_qty actually reflects the variations
+        in stock, both on product and template.
+        """
+        moveObj = self.env["stock.move"]
+        templateObj = self.env["product.template"]
+        supplier_location = self.env.ref("stock.stock_location_suppliers")
+        stock_location = self.env.ref("stock.stock_location_stock")
+        customer_location = self.env.ref("stock.stock_location_customers")
+        uom_unit = self.env.ref("uom.product_uom_unit")
+
+        # Create product template with 2 variant
+        templateAB = templateObj.create(
+            {"name": "templAB", "uom_id": uom_unit.id, "type": "product"}
+        )
+        self.env["product.template.attribute.line"].create(
+            {
+                "product_tmpl_id": templateAB.id,
+                "attribute_id": self.env.ref("product.product_attribute_2").id,
+                "value_ids": [
+                    (
+                        6,
+                        0,
+                        [
+                            self.env.ref("product.product_attribute_value_3").id,
+                            self.env.ref("product.product_attribute_value_4").id,
+                        ],
+                    )
+                ],
+            }
+        )
+
+        # Create product A and B
+        productA = templateAB.product_variant_ids[0]
+        productB = templateAB.product_variant_ids[1]
+
+        # Create a stock move from INCOMING to STOCK
+        stockMoveInA = moveObj.create(
+            {
+                "location_id": supplier_location.id,
+                "location_dest_id": stock_location.id,
+                "name": "MOVE INCOMING -> STOCK ",
+                "product_id": productA.id,
+                "product_uom": productA.uom_id.id,
+                "product_uom_qty": 2,
+            }
+        )
+
+        stockMoveInB = moveObj.create(
+            {
+                "location_id": supplier_location.id,
+                "location_dest_id": stock_location.id,
+                "name": "MOVE INCOMING -> STOCK ",
+                "product_id": productB.id,
+                "product_uom": productB.uom_id.id,
+                "product_uom_qty": 3,
+            }
+        )
+
+        def compare_product_usable_qty(product, value):
+            # Refresh, because the function field is not recalculated between
+            # transactions
+            product.refresh()
```

Comment:
Could you use 
```suggestion
            product.invalidate_recordset()
```

Or
```suggestion
            product.invalidate_recordset(fnames=["immediately_usable_qty"])
```

Reply:
ping @bizzappdev this might fix the tests.